### PR TITLE
Gh 726 add nfl

### LINKED ISSUE #726: Add NFL
This is for contract changes plus any associated backend typescript changes.

See this [tech spec](https://github.com/AugurProject/turbo/blob/dev/specs/nfl-ncaa.md) for details.

The big changes:
- Support for NFL ties.
- Implement listResolvableEvents. The [robert/resolut](https://github.com/AugurProject/turbo/tree/robert/resolut) branch has this code with a similar name.

Work to be done:

- [x] #1037
- [x] #1038
- [x] #1039

### LINKED ISSUE #1079: NFL doesn't draw for spread and OU
I tore out the half-points for all markets for NFL but that's incorrect: only Head to Head can draw.

- [x] dd half-points to spread and OU
- [x] change invalid names to "No Contest" for spread and OU

### PR DESCRIPTION
**Pull request by justinbarry**

### CODE REVIEW

File: packages/comps/src/utils/derived-nfl-dailies.ts [hunk 1]
```diff
@@ -86,7 +117,29 @@ export const getMarketTitle = (
     title = marketTitles.title;
     description = populateHomeAway(marketTitles.description, homeTeam, awayTeam);
   }
+  if (sportsMarketType === 1) {
```
Comment (robert-e-davidson3) [inline packages/comps/src/utils/derived-nfl-dailies.ts:120]: magic number bad

File: packages/comps/src/utils/derived-nfl-dailies.ts [hunk 2]
```diff
@@ -86,7 +117,29 @@ export const getMarketTitle = (
...
+    if (!Number.isInteger(spread)) {
+      spread = Math.trunc(spread);
+    }
+    title = marketTitles.title
+      .replace(NAMING_TEAM.FAV_TEAM, fav)
+      .replace(NAMING_TEAM.UNDERDOG_TEAM, underdog)
+      .replace(NAMING_LINE.SPREAD_LINE, String(spread));
+  }
 
+  if (sportsMarketType === 2) {
```
Comment (robert-e-davidson3) [inline packages/comps/src/utils/derived-nfl-dailies.ts:138]: also

File: packages/smart/addresses.ts [hunk 3]
```diff
@@ -70,10 +70,87 @@ type AddressMapping = {
...
+        address: "0x8524e46E1B0823Ba23454e211e05A4C488020ABC",
+        collateral: "0x5799bFe361BEea69f808328FF4884DF92f1f66f0",
+        ammFactory: "0x7e7FCb06cC1DcBD1E6AFfFC862Ed169A336fB7Ce",
+        fetcher: "0xC5C415cb7eC3ca7dcC26ca7a3fC1126A07122Ec7",
+        description: "nfl",
+        version: "FILL THIS OUT",
+      },
+      {
+        type: "MMALink",
+        subtype: "V1",
```
Comment (robert-e-davidson3) [inline packages/smart/addresses.ts]: should be V2

File: packages/smart/addresses.ts [hunk 4]
```diff
@@ -70,10 +70,87 @@ type AddressMapping = {
...
+        address: "0xA45b74B3544dC6dF01CdbA38558C1E914779Ac8A",
+        collateral: "0x5799bFe361BEea69f808328FF4884DF92f1f66f0",
+        ammFactory: "0x7e7FCb06cC1DcBD1E6AFfFC862Ed169A336fB7Ce",
+        fetcher: "0x2a507840577A7f896C1e190701390c1b08037c61",
+        description: "mma",
+        version: "FILL THIS OUT",
+      },
+      {
+        type: "Crypto",
+        subtype: "V1",
```
Comment (robert-e-davidson3) [inline packages/smart/addresses.ts]: also

**Review by robert-e-davidson3 — DISMISSED** [review]
UI on mumbai is probably broken due to subtype mismatch.

File: packages/comps/src/utils/derived-nfl-dailies.ts (hunk 1 above)
Comment (justinbarry) [reply packages/comps/src/utils/derived-nfl-dailies.ts:120]: @bthaile ?

File: packages/comps/src/utils/derived-nfl-dailies.ts (hunk 2 above)
Comment (justinbarry) [reply packages/comps/src/utils/derived-nfl-dailies.ts:138]: Matches the pattern of the other utils. Will need to be addressed in another refactor.

**Review by bthaile — APPROVED** [review]
Lgtm

**Pull request merged**